# 10_fmtutil_fixes
#
# in texlive-bin/debian/patches/13_fmtutil_fixes the following calls
# to BuildFormat have been removed:
# - BuildFormat/metafont	from bin-metafont.tpm
# - BuildFormat/tex		from bin-tex.tpm
# - BuildFormat/etex		from bin-etex.tpm
# - BuildFormat/pdftex		from bin-pdftex.tpm
#
# This patch does:
# - add the *fixed* versions of these format.XXX.cnf files:
#   . add latex to format.pdflatex.cnf
#   . add pdfetex to format.etex.cnf
#   . remove those two from format.pdftex.cnf
#   . unchanged format.tex.cnf
# - add calls to BuildFormat/tex,etex,pdftex to plain.tpm which contains
#   all the ini files
# - add format.metafont.cnf and the BuildFormat/metafont to metafont.tpm

 texmf-dist/tpm/metafont.tpm       |    3 +++
 texmf-dist/tpm/plain.tpm          |    5 +++++
 texmf/fmtutil/format.etex.cnf     |    2 ++
 texmf/fmtutil/format.latex.cnf    |    1 +
 texmf/fmtutil/format.metafont.cnf |    1 +
 texmf/fmtutil/format.pdftex.cnf   |    1 +
 texmf/fmtutil/format.tex.cnf      |    6 ++++++
 7 files changed, 19 insertions(+)

Index: texlive-base-2007/texmf/fmtutil/format.latex.cnf
===================================================================
--- texlive-base-2007.orig/texmf/fmtutil/format.latex.cnf	2007-01-15 17:24:50.000000000 +0100
+++ texlive-base-2007/texmf/fmtutil/format.latex.cnf	2007-05-21 09:46:10.000000000 +0200
@@ -1 +1,2 @@
 latex           pdftex         language.dat    -translate-file=cp227.tcx *latex.ini
+pdflatex	pdftex		language.dat	-translate-file=cp227.tcx *pdflatex.ini
Index: texlive-base-2007/texmf-dist/tpm/metafont.tpm
===================================================================
--- texlive-base-2007.orig/texmf-dist/tpm/metafont.tpm	2007-01-23 12:46:07.000000000 +0100
+++ texlive-base-2007/texmf-dist/tpm/metafont.tpm	2007-05-21 09:46:10.000000000 +0200
@@ -33,6 +33,9 @@
 texmf-dist/metafont/misc/ztest.mf
 texmf-dist/tpm/metafont.tpm
     </TPM:RunFiles>
+    <TPM:Installation>
+      <TPM:Execute function="BuildFormat" parameter="metafont"/>
+    </TPM:Installation>
     <TPM:Provides>Package/metafont</TPM:Provides>
   </rdf:Description>
 </rdf:RDF>
Index: texlive-base-2007/texmf/fmtutil/format.metafont.cnf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ texlive-base-2007/texmf/fmtutil/format.metafont.cnf	2007-05-21 09:46:10.000000000 +0200
@@ -0,0 +1 @@
+mf		mf-nowin	-		-translate-file=cp227.tcx mf.ini
Index: texlive-base-2007/texmf/fmtutil/format.etex.cnf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ texlive-base-2007/texmf/fmtutil/format.etex.cnf	2007-05-21 13:00:03.000000000 +0200
@@ -0,0 +1,2 @@
+etex            pdftex         language.def    -translate-file=cp227.tcx *etex.ini
+pdfetex         pdftex         language.def    -translate-file=cp227.tcx *pdfetex.ini
Index: texlive-base-2007/texmf/fmtutil/format.pdftex.cnf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ texlive-base-2007/texmf/fmtutil/format.pdftex.cnf	2007-05-21 13:00:39.000000000 +0200
@@ -0,0 +1 @@
+pdftex          pdftex         -       -translate-file=cp227.tcx *pdftex.ini
Index: texlive-base-2007/texmf/fmtutil/format.tex.cnf
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ texlive-base-2007/texmf/fmtutil/format.tex.cnf	2007-05-21 09:47:30.000000000 +0200
@@ -0,0 +1,6 @@
+#
+# Change "tex.ini -> bplain.ini" and "- -> language.dat"
+# if you want babel support in tex. Add -translate-file=cp227.tcx before
+# tex.ini if you want to make all characters directly "printable" for
+# any \write (instead of ^^xy).
+tex             tex             -               tex.ini
Index: texlive-base-2007/texmf-dist/tpm/plain.tpm
===================================================================
--- texlive-base-2007.orig/texmf-dist/tpm/plain.tpm	2007-05-21 09:48:25.000000000 +0200
+++ texlive-base-2007/texmf-dist/tpm/plain.tpm	2007-05-21 09:49:36.000000000 +0200
@@ -44,6 +44,11 @@
 texmf-dist/tex/plain/config/xetex.ini
 texmf-dist/tpm/plain.tpm
     </TPM:RunFiles>
+    <TPM:Installation>
+      <TPM:Execute function="BuildFormat" parameter="tex"/>
+      <TPM:Execute function="BuildFormat" parameter="etex"/>
+      <TPM:Execute function="BuildFormat" parameter="pdftex"/>
+    </TPM:Installation>
     <TPM:Provides>Package/plain</TPM:Provides>
   </rdf:Description>
 </rdf:RDF>
